Topic01/ejercicio3.1

victorja opened the pull request:
Hechos puntos 1,2,3

1)Create a function to get the response from http://api.icndb.com/jokes/random.

2)Replace the button's click event with this new function.

3)Write the response inside the section element.

victorja committed:
Hechos los 3 primeros puntos del ejercicio 3

Create a function to get the response from http://api.icndb.com/jokes/random.

Replace the button's click event with this new function.

Write the response inside the section element.

victorja committed:
actualizado

victorja committed:
agrego if linea 14

victorja committed:
arreglado el error, ahora es rojo

pabloascencao commented on topic1ej3.1/script.js:
```diff
@@ -0,0 +1,25 @@
+		if(http.readyState == 4 && http.status ==200){//si esta todo OK
+			var myObj = JSON.parse(http.response);
+			var list = document.getElementById("appendable");
+				if(list.childNodes.length >1){list.removeChild(list.childNodes[1]);}//si hay impreso un mensaje lo borra
```
esto va a borrar el indice 1 de la lista en el caso que el largo sea mayor a 1.  estas seguro que esto esta bien?

victorja replied:
La idea de la pagina es apretar el boton y que te tire un chiste, si no lo hacia de esa forma se iban acumulando los chistes uno abajo de otro cada vez que apretaba el boton. Por eso ahi el if elimina nodos hijos si los hay, y si no hay ninguno solo agrega el primer hijo, entonces al volver a apretar el boton, elimina el hijo creado anteriormente y crea uno nuevo. 
Volvi a probar el boton y funciona como quiero, cualquier cosa que no haya entendido lo que me queres decir comentame y lo vuelvo a revisar!

pabloascencao replied:
ok, ahi me baje el repo y entendi que era lo que estabas haciendo, lo que pasa es que el comentario que dejaste no era del todo exacto a la funcionalidad.

pabloascencao commented on topic1ej3.1/script.js:
```diff
@@ -0,0 +1,25 @@
+				list.innerHTML += myObj.value.joke;
+
+			}else if(http.status == 403 || http.status == 404){//mensajes de error
+				document.getElementById("appendable").innerHTML += "<p style=\"color:red;\">"+"Error loading the joke"+"</p>"; //mensaje status error
```
es necesario el style inline? no podes usar otra cosa?

victorja committed:
saco el inline style

linea 18, cambio el inline style por metodo que modifica directamente la stylesheet
